src/gpio.rs

```diff
@@ -1,4 +1,39 @@
 //! GPIO and pin configuration
+//!
+//! Interface for the input/output pins.
+//!
+//! `Gpio1` through `Gpio39` represent the *physical* pins of the ESP chip.
+//!
+//! *Logical* pins (compatible with `embedded_hal::digital::InputPin`/`OutputPin`)
+//! are implemented through `PinDriver`.
```

Comment:
Maybe add: "... just like other `embedded_hal` traits are implemented on their corresponding `*Driver` structs and not on the peripherals themselves"

Reply:
You can also sprinkle a mention of the `embedded_hal_async` pin traits as these are also supported by `PinDriver`.